File: minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.java
class Solution {
    public int minStartValue(int[] nums) {
        //add starter=0 to the first index
        //add the first index to the second and so on....
        //keep track of this addition 
        //if there is a case where the sum < 1
        //break out of the current addition and increase starter by one and repeat the steps above
        
        int starter = 1;
        while(true){
            int sum = 0;
            sum+=starter;
            for(int i = 0; i<nums.length; i++){
                sum+=nums[i];
                if(sum<1){
                    break;
                }
            }
            if(sum>=1){
                return starter;
            }
            starter++;
        }
    }
}

File: 1514-minimum-value-to-get-positive-step-by-step-sum/minimum-value-to-get-positive-step-by-step-sum.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        prefix = [0 for v in nums]
        prefix[0] = nums[0]
        i = 0
        last = nums[0]
        while(i < len(nums)):
            if i == 0:
                i+=1
                continue
            last+=nums[i]
            prefix[i] = last
            i+=1
        smallest = min(prefix)
        if smallest > 0:
            return 1
        print(prefix)
        return -1 * smallest + 1

